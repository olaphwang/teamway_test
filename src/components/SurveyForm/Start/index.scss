.start {
  text-align: center;
}

.title {
  margin: 0 auto;
  margin-top: 20px;
  margin-bottom: 15px;
  padding: 0 10px;
  font-size: 20px;
  line-height: 23px;
}

.description {
  margin: 0 auto;
  padding: 0 10px;
  font-size: 14px;
  line-height: 143%;
}

.start-button-container {
  margin: auto;
  width: 50px;
  cursor: pointer;
  border-radius: 5px;
  background-color: aqua;
  padding: 5px;
  margin-top: 30px;
  margin-bottom: 23px;
  display: flex;
  align-items: center;
  justify-content: center;
}
