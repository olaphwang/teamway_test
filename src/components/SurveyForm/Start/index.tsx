import React from 'react';
import styles from './index.scss';

interface StartProps {
  handleNext: () => void;
}
const Start = ({ handleNext }: StartProps) => {
  return (
    <div className={styles.start}>
      <h3 className={styles.title}>
        <strong>Personality trait</strong>
      </h3>

      <p className={styles.description}>
        Build a simple personality test application, that takes 3-5 different questions, maps them into a score and
        produces a personality trait of either Introvert or Extrovert.
      </p>

      <div className={styles['start-button-container']} onClick={handleNext}>
        Start
      </div>
    </div>
  );
};

export default Start;
