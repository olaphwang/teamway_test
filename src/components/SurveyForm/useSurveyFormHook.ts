import { useState, useCallback, useEffect } from 'react';
import questions from '@components/SurveyForm/questions';
import { QuetionsData } from '@components/SurveyForm/types';

const useSurveyFormHook = () => {
  const [questionIndex, setquestionIndex] = useState<number>(-1);
  const [selectedAnswers, setselectedAnswers] = useState({});
  const [quetionsData, setquestionsData] = useState<QuetionsData[]>([]);

  useEffect(() => {
    setquestionsData(questions);
  }, [quetionsData]);

  const handleNext = useCallback(() => {
    if (questionIndex !== quetionsData.length) setquestionIndex((questionIndex) => questionIndex + 1);
  }, [questionIndex, quetionsData]);

  const handlePrev = useCallback(() => {
    if (questionIndex !== 0) setquestionIndex((questionIndex) => questionIndex - 1);
  }, [questionIndex, quetionsData]);

  const handleReStart = useCallback(() => {
    setquestionIndex(0);
    setselectedAnswers({});
  }, [questionIndex]);

  const handleHome = useCallback(() => {
    setquestionIndex(-1);
    setselectedAnswers({});
  }, [questionIndex]);

  return {
    questionIndex,
    setquestionIndex,
    selectedAnswers,
    setselectedAnswers,
    quetionsData,
    setquestionsData,
    handleReStart,
    handleHome,
    handleNext,
    handlePrev,
  };
};

export default useSurveyFormHook;
