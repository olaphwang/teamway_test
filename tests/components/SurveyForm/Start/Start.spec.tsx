import React from 'react';
import renderer from 'react-test-renderer';
import Start from '@components/SurveyForm/Start';
import { renderHook } from '@testing-library/react';
import useSurveyFormHook from '@components/SurveyForm/useSurveyFormHook';

describe('Start component', () => {
  test('should render component properly', () => {
    // when
    const {
      result: { current },
    } = renderHook(() => useSurveyFormHook());
    const { handleNext } = current;
    const componentRenderer = renderer.create(<Start handleNext={() => handleNext()} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
