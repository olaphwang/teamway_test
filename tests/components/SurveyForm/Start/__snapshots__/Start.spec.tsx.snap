// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start component should render component properly 1`] = `
<div
  className="start"
>
  <h3
    className="title"
  >
    <strong>
      Personality trait
    </strong>
  </h3>
  <p
    className="description"
  >
    Build a simple personality test application, that takes 3-5 different questions, maps them into a score and produces a personality trait of either Introvert or Extrovert.
  </p>
  <div
    className="start-button-container"
    onClick={[Function]}
  >
    Start
  </div>
</div>
`;
